zcash_client_sqlite/src/wallet.rs

```diff
@@ -380,16 +381,17 @@ pub(crate) fn get_transparent_receivers<P: consensus::Parameters>(
             })?;
 
         if let Some(taddr) = ua.transparent() {
-            ret.insert(
-                *taddr,
-                AddressMetadata::new(account, DiversifierIndex::from(di_be)),
-            );
+            let di_short = DiversifierIndex::from(di).try_into();
+            if let Ok(di_short) = di_short {
+                if let Some(index) = NonHardenedChildIndex::from_index(di_short) {
+                    ret.insert(*taddr, index);
+                }
+            }
```

Comment:
These two `if let` expressions could silently hide database corruption; neither of them should fail if the transparent address exists, so rather than just filtering out addresses for which there is no result, a failure here must return an error.

Also, I think it would be useful to provide a `TryFrom<DiversifierIndex> for NonHardenedChildIndex` so that you don't have to plumb through the u32 manually here.

Reply:
Or, just use the `to_transparent_child_index` function from https://github.com/zcash/librustzcash/pull/1185/commits/64cd5444af771ea7da91dcbea5d99ac65221e172#diff-190066eb72ebab2517bed56c3fb391dd4ac2b10d9a3283615d57b667f5a784c4R72

Reply:
I thought of reusing that fn, but it isn't public.

Reply:
You can make it `pub(crate)` if you need to.